kafka-rest/src/main/java/io/confluent/kafkarest/v2/KafkaConsumerState.java

```diff
@@ -58,11 +59,8 @@
   private KafkaRestConfig config;
   private ConsumerInstanceId instanceId;
   private Consumer<KafkaKeyT, KafkaValueT> consumer;
-  private ConsumerRecords<KafkaKeyT, KafkaValueT> consumerRecords = null;
-
-  private List<ConsumerRecord<KafkaKeyT, KafkaValueT>> consumerRecordList = null;
-  private int index = 0;
 
+  private Queue<ConsumerRecord<KafkaKeyT, KafkaValueT>> consumerRecords = null;
```

Comment:
If we initialize this to an empty collection, then I think we don't need the null check in `hasNextCached`. Is that right?